build_tests/test_tasks_api.py

```diff
@@ -0,0 +1,332 @@
+            raise Exception(e)
+        
+    def test_get_tasks(self):
+        """Test case for get_tasks
+
+        Get multiple tasks  # noqa: E501
+        """
+        try:
+            # Get multiple tasks
+            tasks = self.api.get_tasks({
+                'assignee': self.USER_GID,
+                'workspace': self.WORKSPACE_GID,
+            })
+            for task in tasks:
+                self.assertIsNotNone(task["gid"])
+                self.assertIsNotNone(task["name"])
+                self.assertIsNotNone(task["resource_type"])
+                self.assertIsNotNone(task["resource_subtype"])
+        except ApiException as e:
+            raise Exception(e)
+
+    def test_get_tasks_with_opt_fields(self):
+        """Test case for get_tasks with opt_fields
+
+        Get multiple tasks with opt_fields # noqa: E501
+        """
+        try:
+            # Get multiple tasks
+            tasks = self.api.get_tasks({
+                'limit': 100,
+                'assignee': self.USER_GID,
+                'workspace': self.WORKSPACE_GID,
+                'opt_fields': "actual_time_minutes,approval_status,assignee,assignee.name,assignee_section,assignee_section.name,assignee_status,completed,completed_at,completed_by,completed_by.name,created_at,created_by,custom_fields,custom_fields.asana_created_field,custom_fields.created_by,custom_fields.created_by.name,custom_fields.currency_code,custom_fields.custom_label,custom_fields.custom_label_position,custom_fields.date_value,custom_fields.date_value.date,custom_fields.date_value.date_time,custom_fields.description,custom_fields.display_value,custom_fields.enabled,custom_fields.enum_options,custom_fields.enum_options.color,custom_fields.enum_options.enabled,custom_fields.enum_options.name,custom_fields.enum_value,custom_fields.enum_value.color,custom_fields.enum_value.enabled,custom_fields.enum_value.name,custom_fields.format,custom_fields.has_notifications_enabled,custom_fields.is_formula_field,custom_fields.is_global_to_workspace,custom_fields.is_value_read_only,custom_fields.multi_enum_values,custom_fields.multi_enum_values.color,custom_fields.multi_enum_values.enabled,custom_fields.multi_enum_values.name,custom_fields.name,custom_fields.number_value,custom_fields.people_value,custom_fields.people_value.name,custom_fields.precision,custom_fields.resource_subtype,custom_fields.text_value,custom_fields.type,dependencies,dependents,due_at,due_on,external,external.data,followers,followers.name,hearted,hearts,hearts.user,hearts.user.name,html_notes,is_rendered_as_separator,liked,likes,likes.user,likes.user.name,memberships,memberships.project,memberships.project.name,memberships.section,memberships.section.name,modified_at,name,notes,num_hearts,num_likes,num_subtasks,offset,parent,parent.created_by,parent.name,parent.resource_subtype,path,permalink_url,projects,projects.name,resource_subtype,start_at,start_on,tags,tags.name,uri,workspace,workspace.name,resource_type",
```

Comment:
Do you end up making an assertion for each below? Or is it just all fields / pasted from sample and then picked some to test?

Reply:
The below loops through each task in the response and asserts the top level. In our `opt_fields` we ask for `assignee.name` but I've only asserted the top level parameters. I didn't want to go too deep into this because this would get really big.